Classes/OAuthClient.php

```diff
@@ -509,4 +516,40 @@ protected function createOAuthProvider(string $clientId, string $clientSecret):
             'requestFactory' => $this->getRequestFactory()
         ]);
     }
+
+    /**
+     * @return void
+     * @throws ORMException
+     * @throws InvalidQueryException
+     */
+    protected function removeExpiredAuthorizations(): void
+    {
+        $query = new Query(Authorization::class);
+        $authorizations = $query->matching($query->lessThan('expires', new \DateTimeImmutable()))->execute();
+        foreach ($authorizations as $authorization) {
+            assert($authorization instanceof Authorization);
+            $this->entityManager->remove($authorization);
+        }
+
+        $this->entityManager->flush();
+    }
+
+    /**
+     * Shuts down this client
+     *
+     * This method must not be called manually – it is invoked by Flow's object
+     * management.
+     *
+     * @return void
+     * @throws InvalidQueryException
+     * @throws ORMException
+     */
+    public function shutdownObject()
+    {
+        $decimals = (integer)strlen(strrchr($this->garbageCollectionProbability, '.')) - 1;
+        $factor = ($decimals > -1) ? $decimals * 10 : 1;
+        if (rand(1, 100 * $factor) <= ($this->garbageCollectionProbability * $factor)) {
```

Comment:
eew, that code is so complicated, it explodes my head :D
Apart from that PHPStorm suggests to use `mt_rand` or even `random_int` here instead - but I guess that's not a major issue in this case

Reply:
- complicated? Might be, stolen from `Neos\Flow\Session.shutdownObject()`…
- `rand()`–I know. But at this point we do not need cryptographically secure randomness. And `random_int()` may throw an exception in case not enough entropy can be gathered, which complicates things for no benefit.

Reply:
ah, right. Didn't know that. Thx!